docs/contracts.rst

```diff
@@ -104,22 +114,115 @@ Each Contract Factory exposes the following methods.
+~~~~~~~~~~~~~~~~
+
+    The Event Log Object is a python dictionary with the following keys:
+
+    * ``args``: Dictionary - The arguments coming from the event.
+    * ``event``: String - The event name.
+    * ``logIndex``: Number - integer of the log index position in the block.
+    * ``transactionIndex``: Number - integer of the transactions index position
+      log was created from.
+    * ``transactionHash``: String, 32 Bytes - hash of the transactions this log
+      was created from.
+    * ``address``: String, 32 Bytes - address from which this log originated.
+    * ``blockHash``: String, 32 Bytes - hash of the block where this log was
+      in. null when its pending.
+    * ``blockNumber``: Number - the block number where this log was in. null
+      when its pending.
+
+
+    .. code-block:: python
+
+        >>> transfer_filter = my_token_contract.eventFilter('Transfer', {'filter': {'_from': '0xdc3a9db694bcdd55ebae4a89b22ac6d12b3f0c24'}})
+        >>> transfer_filter.get_new_entries()
+        [...]  # array of Event Log Objects that match the filter.
+
+        # wait a while...
+
+        >>> transfer_filter.get_new_entries()
+        [...]  # new events since the last call
+
+        >>> transfer_filter.get_all_entries()
+        [...]  # all events that match the filter.
+
+Contract Functions
+------------------
+
+.. py:class:: ContractFunction
+
+The named functions exposed through the :py:attr:`Contract.functions` property are of the ContractFunction type. This class it not to be used directly, but instead through :py:attr:`Contract.functions`.
+
+For example:
+
+    .. code-block:: python
+
+        myContract = web3.eth.contract(address=contract_address, abi=contract_abi)
+        myContract.functions.multiply7(3).call()
+
+:py:class:`ContractFunction` provides methods to interact with contract functions. Positional and keyword arguments supplied to the contract function subclass will be used to find the contract function by signature, and forwarded to the contract function when applicable.
+
+Methods
+"""""""
+
+.. py:method:: ContractFunction.transact(transaction)
 
-    The second portion of the function call ``myMethod(*args, **kwargs)``
+    Execute the specified function by sending a new public transaction.
+
+    Refer to the following invocation:
+
+    .. code-block:: python
+
+        myContract.functions.myMethod(*args, **kwargs).transact(transaction)
+
+    The first portion of the function call ``myMethod(*args, **kwargs)``
     selects the appropriate contract function based on the name and provided
     argument.  Arguments can be provided as positional arguments, keyword
     arguments, or a mix of the two.
 
+    The end portion of this function call ``transact(transaction)`` takes a
+    single parameter which should be a python dictionary conforming to
+    the same format as the ``web3.eth.sendTransaction(transaction)`` method.
+    This dictionary may not contain the keys ``data`` or ``to``.
```

Comment:
I *think* `to` is allowed *if* the method is being called from the contract class (as opposed to a contract instance).